catch/include/hip_test_common.hh

```diff
@@ -311,15 +324,38 @@ template <typename F, typename... Args> void validateArguments(F f, Args...) {
  * @param stream
  * @param packedArgs A list of kernel arguments to be forwarded.
  */
-template <typename... Typenames, typename K, typename Dim, typename... Args>
-void launchKernel(K kernel, Dim numBlocks, Dim numThreads, std::uint32_t memPerBlock,
-                  hipStream_t stream, Args&&... packedArgs) {
+// template <typename... Typenames, typename K, typename Dim, typename... Args>
+// void launchKernel(K kernel, Dim numBlocks, Dim numThreads, std::uint32_t memPerBlock,
+//                   hipStream_t stream, Args&&... packedArgs) {
+// #ifndef RTC_TESTING
+//   validateArguments(kernel, packedArgs...);
+//   kernel<<<numBlocks, numThreads, memPerBlock, stream>>>(std::forward<Args>(packedArgs)...);
+// #else
+//   launchRTCKernel<Typenames...>(kernel, numBlocks, numThreads, memPerBlock, stream,
+//                                 std::forward<Args>(packedArgs)...);
+// #endif
+//   HIP_CHECK(hipGetLastError());
+// }
```

Comment:
This wasn't working - fix implemented below. With the following changes I was able to compile using LLVM17. 

This solution seems not to respect const. Does the original implementation (can't test since it doesn't compile) ?